spec/requests/admin_payment_confirmation_report_upload_spec.rb

```diff
@@ -0,0 +1,83 @@
+require "rails_helper"
+
+RSpec.describe "Admin Payment Confirmation Report upload" do
+  let(:payroll_run) { create(:payroll_run) }
+
+  context "when signed in as a service operator" do
+    let(:admin_session_id) { "some_user_id" }
+    before do
+      sign_in_to_admin_with_role(AdminSession::SERVICE_OPERATOR_DFE_SIGN_IN_ROLE_CODE, admin_session_id)
+    end
+
+    describe "payment_confirmation_report_uploads#new" do
+      it "returns an OK response" do
+        get new_admin_payroll_run_payment_confirmation_report_upload_path(payroll_run)
+
+        expect(response).to have_http_status(:ok)
+      end
+    end
+
+    describe "payment_confirmation_report_uploads#create" do
+      context "the claims in the CSV match the claims of the payroll run" do
+        it "records the values from the CSV against the claims' payments" do
+          payroll_run = create(:payroll_run, claims_count: 2)
+
+          csv = <<~CSV
+            Payroll Reference,Gross Value,Claim ID,NI,Employers NI,Student Loans,Tax,Net Pay
+            DFE00001,487.48,#{payroll_run.claims[0].reference},33.9,38.98,0,89.6,325
+            DFE00002,904.15,#{payroll_run.claims[1].reference},77.84,89.51,40,162.8,534
+          CSV
+
+          file = Rack::Test::UploadedFile.new(StringIO.new(csv), "text/csv", original_filename: "payments.csv")
```

Comment:
worth pulling this up to a `let`?

Reply:
Do you mean a `let` at the `context` level or the `describe` level? If `describe`, there would be some examples in which the `file` is not used. If `context`, do you think that a `let` is better than just creating local variables inside the individual examples?

Reply:
So a `let` is just a memoized helper method used to DRY up specs and make the more readable. It is lazy-evaluated, so if it's not called, nothing happens. It's generally ok to have a `let` that isn't used by all the examples inside a nested block. Not a big deal but I thought it might make the specs easier to read if the mechanics of setting up an upload file wasn't inlined in the specs.

Reply:
Thanks, I'll change it to pull it out as a `let`.